test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -219,6 +219,124 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+	portSet := make(map[string]struct{})
+	for _, port := range beforeOVSPorts {
+		portSet[port] = struct{}{}
+	}
+
+	for _, port := range afterOVSPorts {
+		if _, exists := portSet[port]; !exists {
+			return fmt.Errorf("OVS port %s is missing after restart", port)
+		}
+	}
+
+	return nil
+}
+
+// CheckOVSPortsAfterAgentRestart verifies OVS ports before and after restarting Antrea agent pods.
+func (data *testData) CheckOVSPortsAfterAgentRestart(e2eTestData *antreae2e.TestData, t *testing.T) error {
+	var beforeRestartOVSPorts []string
+	for _, pod := range data.pods {
+		ports, err := data.GetOVSPorts(pod)
+		if err != nil {
+			t.Fatalf("Error getting OVS ports before restart: %v", err)
+		}
+		beforeRestartOVSPorts = append(beforeRestartOVSPorts, ports...)
+	}
+
+	if err := e2eTestData.RestartAntreaAgentPods(90 * time.Second); err != nil {
+		t.Fatalf("Failed to restart Antrea agent pods: %v", err)
+	}
+
+	var afterRestartOVSPorts []string
+	for _, pod := range data.pods {
+		ports, err := data.GetOVSPorts(pod)
+		if err != nil {
+			t.Fatalf("Error getting OVS ports after restart: %v", err)
+		}
+		afterRestartOVSPorts = append(afterRestartOVSPorts, ports...)
+	}
+
+	logs.Infof("Before Restart OVS Ports: %v", beforeRestartOVSPorts)
+	logs.Infof("After Restart OVS Ports: %v", afterRestartOVSPorts)
+
+	if err := data.CheckOVSPorts(beforeRestartOVSPorts, afterRestartOVSPorts); err != nil {
+		t.Fatalf("Mismatch in OVS ports before and after restart: %v", err)
+	}
+	return nil
+}
+
+// RemovePodsAndCheckOVSPorts verifies that OVS ports are removed when pods are deleted.
+func (data *testData) RemovePodsAndCheckOVSPorts(pods []*testPodInfo, waitTime time.Duration) error {
+	pod := data.pods[0] // Select first pod for removal
+	beforeRemovePorts, err := data.GetOVSPorts(pod)
+	if err != nil {
+		return fmt.Errorf("error getting OVS ports before pod removal: %v", err)
+	}
+	logs.Infof("Before Removal - Pod: %s, OVS Ports: %v", pod.podName, beforeRemovePorts)
+
+	// Delete selected pods
+	if err := data.e2eTestData.DeletePodAndWait(waitTime, pod.podName, data.e2eTestData.GetTestNamespace()); err != nil {
+		return fmt.Errorf("failed to delete pod %s: %v", pod.podName, err)
+	}
+
+	// Ensure pods are actually deleted
+	exists, _ := data.CheckPodExists(pod.podName, data.e2eTestData.GetTestNamespace())
+	if exists {
+		return fmt.Errorf("pod %s still exists after deletion", pod.podName)
+	}
+
+	afterRemovePorts, err := data.GetOVSPorts(pod)
+	if err == nil {
+		return fmt.Errorf("error getting OVS ports after pod removal: %v", err)
+	}
+	logs.Infof("After Removal - Pod: %s, OVS Ports: %v", pod.podName, afterRemovePorts)
+
+	// Ensure no stale OVS ports remain
+	if err := data.CheckOVSPorts(beforeRemovePorts, afterRemovePorts); err != nil {
+		logs.Infof("OVS ports has been removed after pod deletion: %v", afterRemovePorts)
+	}
+
+	// Verify that the deleted pod's IPs are released
+	podIPs, err := data.listPodIPs(pod)
```

Comment:
I meant check secondary interface IPs are released to Antrea IPPool, for removed Pods when Agent stopped.
I am fine not to include it to the test case, if it is complex to implement (probably not that hard to get the state from IPPool CR).